### ISSUE TITLE
Optimizer.optimize need to update the input model when it's done

### ISSUE THREAD

**Issue by jason-dai** [intel-analytics/BigDL#395]
The user provides an input model to Optimizer(model, dataset, criterion), and Optimizer.optimize() will return a trained model after it's done:
 1. The local optimizer actually uses the weight of the input model in training
 2. The distributed optimizer will not update the weight of the input model
 3. None of them update other states of the input model (e.g., those in BatchNormalization)

This is very confusing; Optimizer.optimize() should update the input model after it is done, so that the user can use it for prediction.
